### PR TITLE
[IMP] mail, website_slides: use mock server for activity test

### PR DESCRIPTION
**Pull request by phenix-factory**

### CODE REVIEW

File: addons/mail/static/tests/helpers/mock_server.js [hunk 1]
```diff
@@ -307,6 +299,30 @@ MockServer.include({
     // Private Mocked Routes
     //--------------------------------------------------------------------------
 
+    /**
+     * Simulates `activity_format` on `mail.activity`.
+     *
+     * @private
+     * @param {Object} args
+     * @returns {Object}
```
Comment (seb-odoo) [inline addons/mail/static/tests/helpers/mock_server.js:307]: Object[]

**Review by seb-odoo — COMMENTED** [review]
some comments for information, but I'll make the changes and merge it

**Review by seb-odoo — APPROVED** [review]
@robodoo r+

File: addons/mail/static/src/components/activity/tests/activity_tests.js [hunk 2]
```diff
@@ -49,12 +44,22 @@ QUnit.module('activity_tests.js', {
...
-    await this.start();
-    const activity = this.messaging.models['Activity'].create({
+    this.data['res.partner'].records.push({
+        id: 100,
+        activity_ids: [12]
+    });
+    this.data['mail.activity'].records.push({
         id: 12,
-        thread: insert({ id: 42, model: 'res.partner' }),
+        res_id: 100,
```
[the remark this answers is not in the document]
Comment (phenix-factory) [reply addons/mail/static/src/components/activity/tests/activity_tests.js:38]: This is a bit special. In test, if every object as id: 1, it's hard to know who's who inside the data.
I liked the idea to go 1, 10, 100, 1000 etc. some test use it, so I instinctively used them as reference to build my `data[]`.
I have corrected some that where building diff for nothing, but not others with actual diff.

**Pull request closed without merging**